src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyDependencyResolver.cs

```diff
@@ -95,7 +105,16 @@ public AssemblyDependencyResolver(string componentAssemblyPath)
             _assemblyPaths = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
             foreach (string assemblyPath in assemblyPaths)
             {
-                _assemblyPaths.Add(Path.GetFileNameWithoutExtension(assemblyPath), assemblyPath);
+                // Check to see if we have the same managed assembly simple name in the dictionary.
+                // We compare the paths reported and if they're the same, simply ignore the duplicate entries. And only fail if the paths are different
```

Comment:
Are there user errors when we expect this to fail? Would it make sense to simplify it to just `_assemblyPaths.TryAdd(Path.GetFileNameWithoutExtension(assemblyPath), assemblyPath);` ?

Reply:
Yes, my understanding is that we want to continue to throw when the same simple name is used but not for the same assembly with the same full path. We decided to fix this in ADR rather than in the native code given the challenges in getting the casing right

Reply:
Yes - it's possible to run into weird cases where this should fail. One such example:
* On Linux I have a Plugin.dll and Plugin.deps.json
* I create a plugin.dll in the same directory and rename Plugin.deps.json to plugin.deps.json
* I use ADR on plugin.dll

This will result in hostpolicy potentially reporting both Plugin.dll and plugin.dll paths - and we don't know which one to pick for the dictionary - so it should fail.

Reply:
Can the same situation and failure mode happen for regular executables?

Reply:
Maybe - we should try that

Reply:
Decided to stay away from case comparison and take your original suggestion for the fix